apps/designer/app/routes/canvas/$projectId.tsx

```diff
@@ -1,40 +1,33 @@
 import { useLoaderData } from "@remix-run/react";
 import type { LoaderFunction, MetaFunction } from "@remix-run/node";
 import { Canvas } from "~/canvas";
-import { loadCanvasData, type ErrorData, type CanvasData } from "~/shared/db";
-import env, { Env } from "~/env.server";
+import { loadCanvasData, type CanvasData } from "~/shared/db";
 import { ErrorMessage } from "~/shared/error";
 import { sentryException } from "~/shared/sentry";
 import { Root } from "@webstudio-is/react-sdk";
 
-type Data = (CanvasData | ErrorData) & { env: Env };
+type Data = CanvasData | { errors: string };
 
-export const meta: MetaFunction = () => {
-  return { title: "Webstudio canvas" };
+export const meta: MetaFunction = ({ data }: { data: Data }) => {
+  if ("errors" in data) {
+    return { title: "Error" };
+  }
+  const { page } = data;
+  return { title: page.title, ...page.meta };
 };
 
 export const loader: LoaderFunction = async ({ params }): Promise<Data> => {
-  if (params.projectId === undefined) {
-    return { errors: "Missing projectId", env };
-  }
   try {
-    const canvasData = await loadCanvasData({ projectId: params.projectId });
-    return {
-      ...canvasData,
-      env,
-    };
-  } catch (error) {
-    if (error instanceof Error) {
-      const message = `Bad canvas data: \n ${error.message}`;
-      sentryException({ message });
-      return {
-        errors: message,
-        env,
-      };
+    if (params.projectId === undefined) {
+      throw new Error("Missing projectId");
     }
-  }
 
-  return { errors: "Unexpected error", env };
+    return loadCanvasData(params.projectId, params.pageId);
+  } catch (error) {
+    const message = error instanceof Error ? error.message : String(error);
+    sentryException({ message });
```

Comment:
maybe we should make sentry take error instances directly too? Seems like a pattern we will need all the time

Reply:
Yes, and it will be more correct to pass the error object as is to the Sentry API. It's designed to take error object not sure why we pass just the message

Reply:
I think someone misunderstood sentry api then, feel free to change

Reply:
Already did